diff/parse.go

```diff
@@ -612,6 +637,18 @@ func (r *HunksReader) ReadAllHunks() ([]*Hunk, error) {
 	}
 }
 
+// parseOnlyInMessage checks if line is a "Only in {source}: {filename}" and returns source and filename
+func parseOnlyInMessage(line []byte) (bool, []byte, []byte) {
+	re := regexp.MustCompile("Only in ([\x21-\x7E]+): ([\x21-\x7E]+)")
+	if re.Match(line) {
+		slices := re.FindSubmatch(line)
+		if fmt.Sprintf("Only in %s: %s", slices[1], slices[2]) == string(line) {
+			return true, slices[1], slices[2]
+		}
+	}
+	return false, nil, nil
```

Comment:
FYI I did some testing, and it seems the output of BSD diff doesn't do any escaping/etc: For example if you have a file or dir containing `: ` in it, it will just be output verbatim. This makes it impossible to always reconstruct this. For example look at this fun output I constructed:

```shellsession
$ ls internal/trace/foo\ bar internal/trace/hello:\ world
internal/trace/foo bar:
bam      bam: bar

internal/trace/hello: world:
bazz

$ diff -r internal/trace/foo\ bar internal/trace/hello:\ world
Only in internal/trace/foo bar: bam
Only in internal/trace/foo bar: bam: bar
Only in internal/trace/hello: world: bazz
```

This makes me think the regex is too restrictive. I'd suggest using more straightforward non-regex code:
```suggestion
	if !bytes.HasPrefix(line, onlyInMessagePrefix) {
		return false, nil, nil
	}
	line = line[len(onlyInMessagePrefix):]
	idx := bytes.Index(line, []byte(": "))
	if idx < 0 {
		return false, nil, nil
	}
	return true, line[:idx], line[idx+2:]
```

What do you think?

Reply:
I think that situation with complex filename (containing `: `) is much more common than situation when line, which is not a part of the hunk, have passed straightforward non-regex check, but isn't actually an `Only in` message.
So far it probably will be better to have not that strict check.
Have added a test case with your example :)

Reply:
Can we just use this regexp? `^Only in ([^:]+): (.+)$`

This should be simple enough and avoid the problem mentioned above, right?